libdnf5/base/base.cpp

```diff
@@ -82,13 +84,25 @@ namespace libdnf5 {
 static std::atomic<Base *> locked_base{nullptr};
 static std::mutex locked_base_mutex;
 
+static void init_lib_rpm() {
+    static bool lib_rpm_initiated{false};
+    if (!lib_rpm_initiated) {
+        if (rpmReadConfigFiles(nullptr, nullptr) != 0) {
```

Comment:
I don't know what exactly --forcearch does inside libdnf, but since it's obviously about overriding the detected arch: it might make sense to pass it down to rpm too, ie as the second argument of rpmReadConfigFiles().

Reply:
Hm, that might make things more complicated than I assumed :)
Is it possible to call `rpmReadConfigFiles()` multiple times? Currently the function is called very early, even before the `--forcearch` switch is processed (for example because detecting the releasever needs to query rpm database). So if it was possible, we could use the second call of `rpmReadConfigFiles` to "adjust" the architecture during processing of `--forcearch` switch. If this is not possible (which I assume), we need to handle arch and releasever variables differently and initialize rpmdb later.
Then we might run into kind of chicken and egg problem - reading the configuration needs variables (thus to have rpm initialized) and initialization of rpm needs `arch` value which is obtained from configuration (or --forcearch parameter).

Reply:
Calling rpmReadConfigFiles() multiple times is perfectly okay (rpmbuild itself does this), just remember to call rpmFreeRpmrc() for each round.